evm/src/main/java/org/hyperledger/besu/evm/precompile/KZGPointEvalPrecompiledContract.java

```diff
@@ -0,0 +1,154 @@
+
+    String pathToSetup;
+    CKZG4844JNI.Preset bitLength;
+    if (pathToTrustedSetup.isPresent()) {
+      pathToSetup = pathToTrustedSetup.get();
+    } else {
+      InputStream is =
+          KZGPointEvalPrecompiledContract.class.getResourceAsStream(
+              "mainnet_kzg_trusted_setup_4096.txt");
+      try {
+        File jniWillLoadFrom = File.createTempFile("kzgTrustedSetup", "txt");
+        Files.copy(is, jniWillLoadFrom.toPath(), REPLACE_EXISTING);
+        is.close();
+        pathToSetup = jniWillLoadFrom.getAbsolutePath();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    try {
+      BufferedReader setupFile =
+          Files.newBufferedReader(Paths.get(pathToSetup), Charset.defaultCharset());
+      String firstLine = setupFile.readLine();
+      if ("4".equals(firstLine)) {
+        bitLength = CKZG4844JNI.Preset.MINIMAL;
+      } else if ("4096".equals(firstLine)) {
+        bitLength = CKZG4844JNI.Preset.MAINNET;
+      } else {
+        throw new IllegalArgumentException("provided file not a setup for either 4 or 4096 bits");
+      }
+      CKZG4844JNI.loadNativeLibrary(bitLength);
+      try {
+        CKZG4844JNI.loadTrustedSetup(pathToSetup);
+      } catch (RuntimeException alreadyLoaded) {
+        if (alreadyLoaded.getMessage().contains("Trusted Setup is already loaded")) {
+        } else {
+          throw alreadyLoaded;
+        }
+      }
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @VisibleForTesting
+  public void tearDown() {
+    CKZG4844JNI.freeTrustedSetup();
+  }
+
+  @Override
+  public String getName() {
+    return "KZGPointEval";
+  }
+
+  @Override
+  public long gasRequirement(final Bytes input) {
+    // As defined in EIP-4844
+    return 50000;
+  }
+
+  @NotNull
+  @Override
+  public PrecompileContractResult computePrecompile(
+      final Bytes input, @NotNull final MessageFrame messageFrame) {
+
+    if (input.size() != 192) {
+      return new PrecompileContractResult(
+          Bytes.EMPTY,
+          false,
+          MessageFrame.State.COMPLETED_FAILED,
+          Optional.of(ExceptionalHaltReason.PRECOMPILE_ERROR));
+    }
+    // Bytes versionedHash = input.slice(0, 32);
+    Bytes z = input.slice(32, 32);
+    Bytes y = input.slice(64, 32);
+    Bytes commitment = input.slice(96, 48);
+    Bytes proof = input.slice(144, 48);
+
+    Bytes output = Bytes.EMPTY;
+    PrecompileContractResult result;
+    try {
+      boolean proved =
+          CKZG4844JNI.verifyKzgProof(
+              commitment.toArray(), z.toArray(), y.toArray(), proof.toArray());
+
+      // # Return FIELD_ELEMENTS_PER_BLOB and BLS_MODULUS as padded 32 byte big endian values
+      //    return Bytes(U256(FIELD_ELEMENTS_PER_BLOB).to_be_bytes32() +
+      // U256(BLS_MODULUS).to_be_bytes32())
```

Comment:
Is this commented because it will be completed later?

Reply:
bump

Reply:
i left that in because it is from the spec and explains what is going on.